src/java.base/share/classes/java/util/HexFormat.java

```diff
@@ -0,0 +1,1023 @@
+     * return copies of formatters with new parameters.
+     *
+     * @return a hexadecimal formatter with no delimiter and lowercase characters
+     */
+    public static HexFormat of() {
+        return HEX_FORMAT;
+    }
+
+    /**
+     * Returns a hexadecimal formatter with the delimiter and lowercase characters.
+     * The prefix and suffix are empty.
+     * The methods {@link #withDelimiter(String) withDelimiter},
+     * {@link #withUpperCase() withUpperCase}, {@link #withLowerCase() withLowerCase},
+     * {@link #withPrefix(String) withPrefix}, and {@link #withSuffix(String) withSuffix}
+     * return copies of formatters with new parameters.
+     *
+     * @param delimiter a delimiter, non-null, may be empty
+     * @return a {@link HexFormat} with the delimiter and lowercase characters
+     */
+    public static HexFormat ofDelimiter(String delimiter) {
+        return new HexFormat(delimiter, "", "", LOWERCASE_DIGITS);
+    }
+
+    /**
+     * Returns a copy of this {@code HexFormat} with the delimiter.
+     * @param delimiter the delimiter, non-null, may be empty
+     * @return a copy of this {@code HexFormat} with the delimiter
+     */
+    public HexFormat withDelimiter(String delimiter) {
+        return new HexFormat(delimiter, this.prefix, this.suffix, this.digits);
+    }
+
+    /**
+     * Returns a copy of this {@code HexFormat} with the prefix.
+     *
+     * @param prefix a prefix, non-null, may be empty
+     * @return a copy of this {@code HexFormat} with the prefix
+     */
+    public HexFormat withPrefix(String prefix) {
+        return new HexFormat(this.delimiter, prefix, this.suffix, this.digits);
+    }
+
+    /**
+     * Returns a copy of this {@code HexFormat} with the suffix.
+     *
+     * @param suffix a suffix, non-null, may be empty
+     * @return a copy of this {@code HexFormat} with the suffix
+     */
+    public HexFormat withSuffix(String suffix) {
+        return new HexFormat(this.delimiter, this.prefix, suffix, this.digits);
+    }
+
+    /**
+     * Returns a copy of this {@code HexFormat} to use uppercase hexadecimal characters.
+     * The uppercase hexadecimal characters are {@code "0-9", "A-F"}.
+     *
+     * @return a copy of this {@code HexFormat} with uppercase hexadecimal characters
+     */
+    public HexFormat withUpperCase() {
+        return new HexFormat(this.delimiter, this.prefix, this.suffix, UPPERCASE_DIGITS);
+    }
+
+    /**
+     * Returns a copy of this {@code HexFormat} to use lowercase hexadecimal characters.
+     * The lowercase hexadecimal characters are {@code "0-9", "a-f"}.
+     *
+     * @return a copy of this {@code HexFormat} with lowercase hexadecimal characters
+     */
+    public HexFormat withLowerCase() {
+        return new HexFormat(this.delimiter, this.prefix, this.suffix, LOWERCASE_DIGITS);
+    }
+
+    /**
+     * Returns the delimiter between hexadecimal values in a formatted byte array.
```

Comment:
"... in a formatted byte array", I find this wording a little hard to follow. The delimiter, prefix, suffix apply to *formatted hexadecimal strings*.

Reply:
It is the byte array that is formatted, the result is a hexadecimal string.
It helps make a stronger distinction between the byte array functions and the primitive functions.

Reply:
> It is the byte array that is formatted, the result is a hexadecimal string.

I don't understand. How is the byte array formatter?   Do we have "formatted byte arrays" and "unformatted byte arrays"?  Are they formatted somehow with prefix, delimiter, suffix ?

Reply:
The question is how to describe the resulting string. 
You are correct that the class javadoc starts off using the terms "hexadecimal string" and "formatted hexadecimal string" but does not follow through.